src/loader/bgw_launcher.c

```diff
@@ -471,35 +532,27 @@ static void
 message_restart_action(HTAB *db_htab, BgwMessage *message, VirtualTransactionId vxid)
 {
 	DbHashEntry *db_he;
-	bool		found;
-	bool		worker_registered;
-	pid_t		worker_pid;
 
-	db_he = hash_search(db_htab, &message->db_oid, HASH_ENTER, &found);
-	if (found)
+	db_he = allocate_scheduler(db_htab, message->db_oid);
```

Comment:
Should we really call `allocate_scheduler` here, given that it creates a new scheduler if one doesn't exist? IMO, if we are doing a "restart" we should only do `HASH_FIND` and error out if no scheduler was found that we could restart.